### PR TITLE
(DOCSP-23554): [Admin API] Sample & Validate Documents

### PR DESCRIPTION
**Pull request by nlarew**
### Jira

- (DOCSP-23554): [Admin API] Sample & Validate Documents

### Staged Changes (Requires MongoDB Corp SSO)

- [Admin API > Schemas > Sample & Validate Documents](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/admin-validate_documents/admin/api/v3/#operation/adminValidateDocuments)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/openapi-admin-v3.yaml [hunk 1]
```diff
@@ -4749,6 +4838,174 @@ components:
           $ref: "#/components/schemas/GraphQLValidationAction"
         write_validation_level:
           $ref: "#/components/schemas/GraphQLValidationLevel"
+    ValidationError:
+      description: An ejson schema validation error.
```
Comment (mongodben) [inline source/openapi-admin-v3.yaml:4842]:
```suggestion
      description: An EJSON schema validation error.
```

File: source/openapi-admin-v3.yaml [hunk 2]
```diff
@@ -4749,6 +4838,174 @@ components:
...
+      const: "const"
+      description: A field has an invalid constant value.
+    EnumValidationError:
+      type: string
+      const: "enum"
+      description: A field has an invalid enum value.
+    NumberAnyOfValidationError:
+      type: string
+      const: "number_any_of"
+      description: A field does not match any schema in anyOf.
```
Comment (mongodben) [inline source/openapi-admin-v3.yaml:4904]: i'm confused here by what you mean by "anyOf". ditto for "oneOf", "allOf" in the next 2 errors. could you clarify a bit more or link to whatever this is?

File: source/openapi-admin-v3.yaml [hunk 3]
```diff
@@ -4749,6 +4838,174 @@ components:
...
+      const: "format"
+      description: A field contains a string with an invalid format.
+    ConditionThenValidationError:
+      type: string
+      const: "condition_then"
+      description: A value is not valid under a truthy conditional.
+    ConditionElseValidationError:
+      type: string
+      const: "condition_else"
+      description: A value is not valid under a falsy conditional.
```
Comment (mongodben) [inline source/openapi-admin-v3.yaml:5008]: for truthy/falsy, mught be worth linking somewhere for definitions

File: source/openapi-admin-v3.yaml (hunk 2 above)
Comment (nlarew) [reply source/openapi-admin-v3.yaml:4905]: It's JSON schema polymorphism & inheritance - added links to the JSON schema site.

File: source/openapi-admin-v3.yaml [hunk 4]
```diff
@@ -605,6 +605,88 @@ paths:
...
+                collection_name:
+                  type: string
+                  description: The name of the collection to validate.
+                from_schema:
+                  type: object
+                  description: The EJSON schema to validate sampled documents against.
+                query:
+                  type: object
+                  default: {}
+                  description: A MongoDB query for the collection. Use this to include a specific subset of documents in the sample.
```
Comment (mongodben) [inline source/openapi-admin-v3.yaml]:
```suggestion
                  description: A MongoDB query for the collection. Use this to include a specific subset of documents in the collection for the sample.
```

and could you provide a link for the text "MongoDB Query"?

File: source/openapi-admin-v3.yaml (hunk 2 above)
Comment (mongodben) [reply source/openapi-admin-v3.yaml]: where is the link?

**Review by mongodben — APPROVED** [review]
lgtm w 1 comment remaining to address

**Pull request merged**